apps/dashboard/src/components/workflow-editor/steps/configure-step-form.tsx

```diff
@@ -57,10 +57,16 @@ export const ConfigureStepForm = (props: ConfigureStepFormProps) => {
     navigate(buildRoute(ROUTES.EDIT_WORKFLOW, { environmentSlug: environment.slug!, workflowSlug: workflow.slug }));
   };
 
+  const defaultValues = {
+    name: step.name,
+    stepId: step.stepId,
+  };
+
   const form = useForm<z.infer<typeof stepSchema>>({
-    defaultValues: {
-      name: step.name,
-      stepId: step.stepId,
+    defaultValues,
+    values: {
+      ...defaultValues,
+      ...step.controls.values,
```

Comment:
According to some posts `values` should be used for values that depend on async return and `defaultValues` on static fields, but the react-hook-form docs states various other things, quite confusing.

Reply:
TBH I am not sure I understand the fix. If it works it works but I suggest adding a comment in the code.